4.sc_Morphology_Neural_Network_MLP_Model/1.model_training/notebooks/regression_call.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#SBATCH --nodes=1
+#SBATCH --ntasks=16
+#SBATCH --mem=300G
+#SBATCH --partition=amem
+#SBATCH --qos=mem
+#SBATCH --time=24:00:00
+#SBATCH --output=sample-%j.out
+
+# module load anaconda
+
+# conda activate Interstellar
+
+cell_types=( SHSY5Y )
+control_names=( DMSO_0.100_DMSO_0.025 )
+treatment_names=( LPS_100.000_DMSO_0.025 Thapsigargin_10.000_DMSO_0.025 LPS_10.000_DMSO_0.025 LPS_1.000_DMSO_0.025 LPS_0.100_DMSO_0.025 LPS_0.010_DMSO_0.025 Thapsigargin_1.000_DMSO_0.025 )
+model_names=( DMSO_0.025_vs_LPS_100 DMSO_0.025_vs_Thapsigargin_10 DMSO_0.025_vs_LPS_10 DMSO_0.025_vs_LPS_1 DMSO_0.025_vs_LPS_0.1 DMSO_0.025_vs_LPS_0.01 DMSO_0.025_vs_Thapsigargin_1 )
+shuffles=( True False )
+
+for cell_type in "${cell_types[@]}"; do
+    for control_name in "${control_names[@]}"; do
+        for treatment_name in "${!treatment_names[@]}"; do
+            for shuffle in "${shuffles[@]}"; do
```

Comment:
Seeing how this pattern is repeated in multiple files made me wonder if a block of Python code could help generate the contents found here, meaning you could create the bash files from Python instead of having to write bash for-loops. This might help conserve your development velocity for Pythonic context / thinking rather than bash context. I can see things either way here, so this isn't a change requirement.

Reply:
bash doesn't both me and I have since in newer scripts iterated over a total array so that each combination can be sent to a different core/node on the HPC but I haven't updated here yet.